### ISSUE TITLE
Tracer.capture_method decorator does not preserve function type hints

### ISSUE THREAD

**Issue by zlalvani** [awslabs/aws-lambda-powertools-python#465]
**What were you trying to accomplish?**
I want to use the `@tracer.capture_method` decorator for detailed function level tracing however it's currently unusable because it disables all errors when linting with `mypy`.

## Expected Behavior
See code example below

## Current Behavior
Function signatures (argument and return types) are not preserved when decorated with `@tracer.capture_method` and linting with `mypy`. 

## Possible Solution
Mypy can be forced to look at the decorated function signature: https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators

## Steps to Reproduce (for bugs)

1. Use code sample below
2. Run mypy (tested on `mypy==0.782`)

## Environment

* **Powertools version used**: `aws-lambda-powertools==1.10.2`
* **Packaging format (Layers, PyPi)**: PyPi

> [How to enable debug mode](https://awslabs.github.io/aws-lambda-powertools-python/#debug-mode)**

```python
tracer = Tracer()

@tracer.capture_method
def foo(bar: str) -> str:
    return bar

x = foo('baz')
x.get('this is not a dict')  # should cause mypy error
y = foo(123)  # should cause mypy error
```

**Comment by heitorlessa** [comment]
Hey @zlalvani thanks a lot for raising this. While we've had numerous MyPy issues fixed we don't yet have full support for it -- Roadmap item that you can help us +1 to prioritize: https://github.com/awslabs/aws-lambda-powertools-roadmap/issues/3

I'm unsure how to fix this atm as I had thought `functools.wraps` would ensure some level of correctness here (not at wrapt level, tho) -- This is that part of the code: https://github.com/awslabs/aws-lambda-powertools-python/blob/8724294013e3157de9478f90e7c1283a952de54f/aws_lambda_powertools/tracing/tracer.py#L607

I've been meaning to refactor `capture_method` as it's the most complex logic we have in the codebase, so it'd be a good exercise nevertheless to get this improved.

I'd appreciate if you have any guidance on how we can satisfy MyPy in a backwards compatible way - happy to provide any guidance if you'd like to try contributing it.

Thank you!

**Comment by heitorlessa** [comment]
Updating here to let you know the fix will be available in the next release

**Comment by heitorlessa** [comment]
Just merged the fix for synchronous functions - Async however requires the new ParamSpec only available in 3.10, so I'll wait for more customers using mypy + async fn as it'll increase the package size by 240K.

This will be available in next week's release (1.18.0) -- I'll ping back here once it's released.

Thank you again for flagging it @zlalvani

**Comment by heitorlessa** [comment]
Hi @zlalvani - This is now fixed in 1.18.0

More details in the release notes: https://github.com/awslabs/aws-lambda-powertools-python/releases/tag/v1.18.0

**Comment by zlalvani** [comment]
Awesome thanks for the fix! @heitorlessa
